examples/gtc_final_report_matrix.py

```diff
@@ -0,0 +1,146 @@
+import sys
+import argparse
+import os
+from datetime import datetime
+from IlluminaBeadArrayFiles import BeadPoolManifest, GenotypeCalls, code2genotype
+from pandas import DataFrame
+from numpy import around
+
+"""
+This example will allow the user to create matrix report files identical to the ones exported from GenomeStudio
+by chosing the genotype with or without the genotype scores.
+"""
+
+def build_dict(matrix = None, samplename = None, names = None, genotypes = None, genotype_scores = None):
+    if genotype_scores is not None:
+        for (name, genotype, genotype_score) in zip(names, genotypes, genotype_scores):
+            if samplename in matrix:
+                matrix[samplename][name] = genotype + "|" + str(genotype_score)
+            else:
+                matrix[samplename] = {}
+                matrix[samplename][name] = genotype + "|" + str(genotype_score)
+    else:
+        for (name, genotype) in zip(names, genotypes):
+            if samplename in matrix:
+                matrix[samplename][name] = genotype
+            else:
+                matrix[samplename] = {}
+                matrix[samplename][name] = genotype
+    return matrix
+
+delim = "\t"
+NUM_ARGUMENTS = 6
+
+parser = argparse.ArgumentParser("Generate a final report from a directory of GTC files")
+parser.add_argument("manifest", help="BPM manifest file")
+parser.add_argument("gtc_directory", help="Directory containing GTC files")
+parser.add_argument("output_file", help="Location to write report")
+parser.add_argument("--forward", help="python gtc_final_report_matrix.py <path_to_manifest> <path_to_gtc_directory> <path_to_output_file> --forward 1, print matrix with forward alleles")
```

Comment:
Should there be an enumerated format option here?

Reply:
I don't see any harm in explicitly spelling out the arguments in the parser as long as we are not expecting many more cases. But if you like I can use an enumerator for the options. I would think the trade-off in better readability not that big though.